l1-contracts/contracts/bridge/ntv/NativeTokenVault.sol

```diff
@@ -0,0 +1,419 @@
+        ASSET_ROUTER.setAssetHandlerAddressThisChain(bytes32(uint256(uint160(_nativeToken))), address(this));
+        tokenAddress[assetId] = _nativeToken;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            FINISH TRANSACTION FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    ///  @inheritdoc IAssetHandler
+    /// @notice Used when the chain receives a transfer from L1 Shared Bridge and correspondingly mints the asset.
+    /// @param _chainId The chainId that the message is from.
+    /// @param _assetId The assetId of the asset being bridged.
+    /// @param _data The abi.encoded transfer data.
+    function bridgeMint(
+        uint256 _chainId,
+        bytes32 _assetId,
+        bytes calldata _data
+    ) external payable override onlyAssetRouter whenNotPaused {
+        // Either it was locked before, therefore is not zero, or it is sent from remote chain and standard erc20 will be deployed
+        // Note: if the chainBalance has not been migrated and is 0, then the assetId check will fail as the token's origin chain is not chain that sent the message.
+        // Note: after interop is implemented this will not work.
+        address token = tokenAddress[_assetId];
+        address receiver;
+        uint256 amount;
+        if (chainBalance[_chainId][token] > 0) {
+            (receiver, amount) = _bridgeMintNativeToken(_chainId, _assetId, _data);
+        } else {
+            (receiver, amount) = _bridgeMintBridgedToken(_chainId, _assetId, _data);
+        }
+        // solhint-disable-next-line func-named-parameters
+        emit BridgeMint(_chainId, _assetId, receiver, amount);
+    }
+
+    function _bridgeMintBridgedToken(
+        uint256 _originChainId,
+        bytes32 _assetId,
+        bytes calldata _data
+    ) internal virtual returns (address receiver, uint256 amount) {
+        // Either it was locked before, therefore is not zero, or it is sent from remote chain and standard erc20 will be deployed
+        address token = tokenAddress[_assetId];
+        bytes memory erc20Data;
+        address originToken;
+        // slither-disable-next-line unused-return
+        (, receiver, originToken, amount, erc20Data) = DataEncoding.decodeBridgeMintData(_data);
+
+        if (token == address(0)) {
+            token = _ensureTokenDeployed(_originChainId, _assetId, originToken, erc20Data);
+        }
+
+        IBridgedStandardToken(token).bridgeMint(receiver, amount);
+        emit BridgeMint(_originChainId, _assetId, receiver, amount);
+    }
+
+    function _bridgeMintNativeToken(
+        uint256 _originChainId,
+        bytes32 _assetId,
+        bytes calldata _data
+    ) internal returns (address receiver, uint256 amount) {
+        // Either it was locked before, therefore is not zero, or it is sent from remote chain and standard erc20 will be deployed
```

Comment:
the comment is incorrect, erc20 will not be deployed in case the address for the native token is 0